Adding first draft of sourcecred glossary

vsoch opened the pull request:
This will add an early draft of the sourcecred glossary, to address #5. Specifically, I've taken the terms that have been shared across various platforms, and put them into a glossary.md that is broken into two sections, User Glossary and Community Glossary. For each, I thought it would make sense to:

 - start with a definition
 - for longer texts, walk the user through the content by way of questions
 - end with references and "in the wild" if appropriate

And I also acknowledge that these terms have many resources out in the wild, but these should be "sourcecred specific," and this is why I include references at the bottom of each. We don't intend to re-write the internet in a GitHub markdown file :)  My thinking is that when this is rendered into a web UI, we can do some javascript tricks to hide specific sections and have the user click to expand if the page gets very long.

Also notably, I've taken several posts from discourse by @decentralion and @Beanow and refactored them here (I hope that's okay) along with linking the original sources. 

## How do I review this?

 - It's hugely likely that some (or even most) of my efforts at definition could be improved upon - please give comments here! 
 - The focus of this PR arguably should be mostly on the general structure / organization.
 - Given the previous point, we probably don't want to suggest entirely new terms. Leave that for separate PRs.
 - Feedback on the general tone of the document would make sense too.

And of course anything else that you might think of! I think having this glossary under version control is a good idea.

Signed-off-by: Vanessa Sochat <vsochat@stanford.edu>

decentralion commented on community/glossary.md:
```diff
@@ -0,0 +1,195 @@
+
+> It would probably be best to split this into two PRs, one which adds the badges and one which adds the colors, as color choices are always an ample source of bikeshedding :)
+
+In the above statement, he recognizes that contributors are likely to get distracted
```
nit: i use they/them pronouns

vsoch replied:
So sorry about that! I definitely know this, my fingers just subconsciously typed it. I'll try harder in the future to be careful.

Beanow commented:
> Also notably, I've taken several posts from discourse by @decentralion and @Beanow and refactored them here (I hope that's okay) along with linking the original sources

Yeah, feel free on my part!

> Structurally, I think we might want to be less ambitious with the glossary. With the level of detail that we're going into here, it's becoming a full guide on core concepts. Which is great, in that we definitely need those guides--e.g. we should have a full guide on what Cred is, and which dives into how it's computed. But I think the glossary should have smaller scope, and link out to those guides/articles/references as appropriate. In which case I'd advocate for shorter definitions like:

Agree with this. The glossary itself I would keep really light on details and link to pages with more info.
Bigger topics like Cred, Grain, Trust levels, would make sense to me if they had their own `concepts/cred.md` sort of document to link to.

---

Here are the goals that I imagine the glossary should have.
(Feel free to challenge me on these though)

**To serve as a starting point.**
You're not sure what a certain term means. Searching the forums you might get a handful of matches. Do they do a good job of explaining? No idea. When it's in the glossary you can be sure it links to an "officially blessed" explanation.

**Be easy to add to.**
I would rather have _some_ explanation here than none.
And would like if all community members feel able to quickly add a term they were confused about, as they bump into them. A link to an existing forum post is much easier than trying to summarize a definition.

**Allow refining.**
A number of these concepts. Such as Champions. Don't have a solid working definition yet :sweat_smile:.
There's a partial attempt [here](https://discourse.sourcecred.io/t/champions-lead-the-charge-on-initiatives/403). There's the original [here](https://discourse.sourcecred.io/t/about-champions-and-heroes/291).
Others took a stab at defining it too, like [here](https://discourse.sourcecred.io/t/champions/527) and [here](https://discourse.sourcecred.io/t/draft-initiatives-guide/525).
Which brings us to even more terms, like [DRIs](https://about.gitlab.com/handbook/people-group/directly-responsible-individuals/).
But I think a straightforward definition is still pending.

So I think we should allow for this in the glossary. Blessing a link to a forum post with the idea of "for now, this is the best place to start". Possibly updating it later if something preferable is written. For example as a core concept page in this repository :]

**Should scale.**
My thought is that we shouldn't be worried even if this grows into a few dozens of terms.
So in the interest of that, I would make entries even smaller than @decentralion suggested.

Either no explanation, completely deferring to the link. Or a one-liner when it makes sense.
> Cred is a unit, used by SourceCred as a metric to mean value / importance.

vsoch commented:
okay updated, and some quick notes!

>  Structurally, I think we might want to be less ambitious with the glossary. With the level of detail that we're going into here, it's becoming a full guide on core concepts. Which is great, in that we definitely need those guides--e.g. we should have a full guide on what Cred is, and which dives into how it's computed. But I think the glossary should have smaller scope, and link out to those guides/articles/references as appropriate. In which case I'd advocate for shorter definitions like:

This is a great idea (and @Beanow thinks so too!). I've shortened all the definitions, and added a folder of concepts pages.  I was thinking that we could hide some of the length by collapsing terms, but I like the idea of having single pages for more modular development, and not needing to render a single, massive page. Some of them do get quite a bit lengthy. I've also updated the "cred" definition to be the one that you provided.

> I'd like to also suggest changing this PR so that the guide doesn't have any real content (just lorem ipsem style entries), and just puts forward the structure. That way we can approve it on structure per your request, and give each glossary entry its own PR for specific review.

If I had not spent quite a bit of time on writing or re-phrasing these definitions, I'd say okay. At this point to delete everything and either need to re-create the PR or lose all the work that I did would be frustrating, and it feels artificial as well to replace actual thought-out content with Lorem ipsum. I apologize that it's not ideal to review multiple terms at once, maybe in the future for the issues that scope out the work, we should do a better job to say specifically what is wanted. For #5 I saw the sections, and also the links and listing of terms, so I naturally intuited opening a PR with those sections and terms would be welcome. I can also try to ask for clarification / start discussion on those issues before starting work - I tend to try and "get it done" without spending extra time asking for permission, so that's my fault as well.

> A number of these concepts. Such as Champions. Don't have a solid working definition yet 

If it would have been okay to have lorem ipsum, I would say it's okay to have a good first shot at some definition for this first go. Please feel free to make tweaks on the current definitions that I've added - you guys wrote most of it for cred / grain / champion so I hope you still like your previous words!

decentralion commented on community/concepts/grain.md:
```diff
@@ -0,0 +1,59 @@
+### Grain
```
I should probably take responsibility for this concept -- the posts that this document are based on are fairly out of date, might be cleaner if I write it from scratch.

decentralion commented on community/concepts/deep-then-wide.md:
```diff
@@ -0,0 +1,30 @@
+# deep then wide
```
(as per review comment in the glossary) -- i'm nto sure deep-then-wide needs a concept here.

vsoch replied:
It was added on the discourse by @Beanow so I thought it might be worth explaining. I can definitely see that someone might not get what is meant, so that alone would justify adding it (for me).

decentralion commented on community/glossary.md:
```diff
@@ -0,0 +1,79 @@
+ - [What is bikeshedding](https://css-tricks.com/what-is-bikeshedding/)
+ - [SourceCred bikeshedding concept](concepts/bikeshedding.md)
+
+### deep then wide
```
Not sure that deep-then-wide merits a glossary entry. It was really influential in orienting the SC strategy around mid 2019, at this point though I feel like the term doesn't come up so much in our discussions.

vsoch replied:
@Beanow and others, what do you think?

vsoch commented on community/glossary.md:
```diff
@@ -0,0 +1,79 @@
+
+SourceCred is a young project, and as we grow it, we can either go "wide" (encourage
+adoption by a lot of projects) or go "deep" (focus on intense adoption by fewer projects).
+While both are meaningful goals, in the case of SourceCred, it is logical to go "wide"
```
oops, fixed locally!

vsoch commented:
Updates (and separation of the concepts into separate PRs) coming soon!

vsoch commented:
ruhroh, looks like I haven't set up GPG on here yet.

Beanow commented on community/glossary.md:
```diff
@@ -0,0 +1,79 @@
+
+Cred is a metric that describes how important a contribution (or contributor) was to a particular project. Using SourceCred's own documentation as an example: the pull request that added this term to the glossary might have 10 cred, while a different pull request that fixes a typo might have only one cred. That would indicate that the first pull request was ten times as important. Cred "flows" be ...
+
+Cred is project-specific, and is computed based on the project's own [contribution graph](). For more details, see the [deep dive guide on cred](https://discourse.sourcecred.io/t/a-gentle-introduction-to-cred/20) or the [concept guide here](concepts/cred.md).
```
Nit: link missing for contribution graph

vsoch replied:
Just want to verify before I add it - this one? https://sourcecred.io/cred/timeline/@sourcecred/

Beanow replied:
Yeah I think for now that's the best one :]
([New UI](https://discourse.sourcecred.io/t/sourcecred-beta-roadmap/422/3) pending)

vsoch replied:
All set.

Beanow commented on community/glossary.md:
```diff
@@ -0,0 +1,79 @@
+
+### Grain
+
+A big part of SourceCred’s vision is the desire to create project-specific cryptoassets, called "Grain" that represent contribution to or support of a project. These cryptoassets hold promise to make open-source projects financially sustainable which is currently an unsolved problem. You can read more about grain [here](concepts/grain.md).
```
Think it would be clearer to start with a definition.
"Grain is ___"

vsoch replied:
Fixed locally.

vsoch commented:
> Structure wise, it might be useful to generate a table of contents. Though if we switch from github rendering to a static site generator this might be included in the theme.

@Beanow I had this exact same thought, and started adding one at the top! I removed it because I realized that this work would be quickly followed up with a template to render it, and (for easily 9/10 of the rendered sites I've done on here) they have included a nice TOC. So - I can add one if you think it's absolutely necessary between the "just markdown" and "rendered site" phases of the docs repository, otherwise we should have one sooner than we both think.

> I do find the names non-obvious.
> User glossary
> Community glossary

I used these from @decentralion suggestion in #5 - I think the general idea is that user terms == using SourceCred and community terms == things I might encounter on GitHub, discord, or other places interacting with the community.

> The actual distinction seems good, just no clue what it is from just the title. Would SourceCred glossary and development glossary make sense?

I think SourceCred glossary is more clear than user glossary, but I think we should stick with something like "Community" over development, because "development" sends a message that "the community is mostly developers" and I don't think we want to send that message."

Beanow commented:
> I had this exact same thought, and started adding one at the top! I removed it because I realized that this work would be quickly followed up with a template to render it,

Happy to skip this for now to let a template do this later :]

> I used these from @decentralion suggestion in #5 - I think the general idea is that user terms == using SourceCred and community terms == things I might encounter on GitHub, discord, or other places interacting with the community.

Yeah that's how I interpreted them too when looking into it. Though without context I wouldn't have figured that out.

> but I think we should stick with something like "Community" over development, because "development" sends a message that "the community is mostly developers" and I don't think we want to send that message."

Yeah that's fair. I tend to think of development in a very broad sense, but that wouldn't be obvious either. From #5 I like the other suggestion "Contributor Glossary". That's what I had in mind for development in the broad sense. Contributing.

I like this over community, because Community glossary suggests two things to me:
- These are somehow terms unique to our community (while usually it's adopted terms)
- You'll need these to be part of the community (imo not true, you may need them for contributing)

So I'd prefer, SourceCred glossary and Contributor glossary.

vsoch commented:
@Beanow updated to Contributor Glossary.
